packages/admin-ui/src/lib/core/src/components/app-shell/app-shell.component.ts

```diff
@@ -70,17 +70,23 @@ export class AppShellComponent implements OnInit {
         this.uiLanguageAndLocale$
             .pipe(
                 take(1),
-                switchMap(([currentLanguage, currentLocale]) =>
-                    this.modalService.fromComponent(UiLanguageSwitcherDialogComponent, {
+                switchMap(([currentLanguage, currentLocale]) => {
+                    const defaultLocale = getAppConfig().defaultLocale;
+                    const modalCurrentLocale = currentLocale ?? defaultLocale;
+
+                    const defaultLanguage = getAppConfig().defaultLanguage;
+                    const modalCurrentLanguage = currentLanguage ?? defaultLanguage;
```

Comment:
I don't understand this part - 

The setting of the `currentLanguage` is already handled here: https://github.com/vendure-ecommerce/vendure/blob/08ad982c6ef31fd53c5d95937c83ae8e313b9f32/packages/admin-ui/src/lib/core/src/core.module.ts#L94-L97

and the currentLocale is set here: https://github.com/vendure-ecommerce/vendure/blob/2cf8f094a98c65274dd4eb0bc26f54c84301bc16/packages/admin-ui/src/lib/core/src/data/client-state/client-defaults.ts#L8

why does this need to be repeated here?

Reply:
done ✅
Yeah, you are right 👌
calculation and assignment of  `modalCurrentLocale` & `modalCurrentLanguage` removed